plugins/notifications-backend/src/database/DatabaseNotificationsStore.ts

```diff
@@ -87,24 +89,63 @@ export class DatabaseNotificationsStore implements NotificationsStore {
   };
 
   private mapToNotifications = (rows: any[]): Notification[] => {
-    return rows.map(row => ({
-      id: row.id,
-      user: row.user,
-      created: new Date(row.created),
-      saved: row.saved,
-      read: row.read,
-      updated: row.updated,
-      origin: row.origin,
-      payload: {
-        title: row.title,
-        description: row.description,
-        link: row.link,
-        topic: row.topic,
-        severity: row.severity,
-        scope: row.scope,
-        icon: row.icon,
-      },
-    }));
+    let order = 0;
+    const res = [
+      ...rows
+        .reduce((acc, row) => {
+          const metadata = row.name
+            ? {
+                name: row.name,
+                value: row.value,
+                type: row.type,
+              }
+            : undefined;
+          if (acc.has(row.id)) {
+            if (metadata) {
+              acc.get(row.id).notification.payload.metadata?.push(metadata);
+            }
+          } else {
+            order += 1;
+            acc.set(row.id, {
+              order,
```

Comment:
Oh hang on. The order isn't needed as far as I know; `.values` returns entries in the order that they were added to the map in the first place.

Reply:
There is a problem because of **unionAll**...
I faced it when on the face of fixing unit tests, and found that have to add this logic 😅